File: go/customer_type.go
// this file is @generated
package meteroid

// Company vs. individual (B2C). Defaults to `COMPANY`.
type CustomerType string

// Known values of CustomerType.
//
// The API may add new values over time; unknown values round-trip unchanged.
const (
	CustomerTypeCompany    CustomerType = "COMPANY"
	CustomerTypeIndividual CustomerType = "INDIVIDUAL"
)

// AllCustomerTypeValues lists every CustomerType value known to this SDK version.
var AllCustomerTypeValues = []CustomerType{
	CustomerTypeCompany,
	CustomerTypeIndividual,
}

// String returns the wire representation of the value.
func (e CustomerType) String() string {
	return string(e)
}

// IsKnown reports whether the value is one this SDK version knows about.
func (e CustomerType) IsKnown() bool {
	for _, known := range AllCustomerTypeValues {
		if e == known {
			return true
		}
	}
	return false
}

File: python/meteroid/models/customer_type.py
# this file is @generated
import enum


class CustomerType(str, enum.Enum):
    """Company vs. individual (B2C). Defaults to `COMPANY`."""

    COMPANY = "COMPANY"
    INDIVIDUAL = "INDIVIDUAL"

    def __str__(self) -> str:
        return str(self.value)
